azure-pipelines/end-to-end-tests-prelude.ps1

```diff
@@ -12,6 +13,7 @@ $commonArgs = @(
     "--x-buildtrees-root=$buildtreesRoot",
     "--x-install-root=$installRoot",
     "--x-packages-root=$packagesRoot",
+    "--downloads-root=$downloadsRoot",

```

Comment:
I would prefer only redirecting the downloads-root in the `clean-after-build` tests, since the rest of the tests can use cached downloads just fine.